### PR TITLE
Add Unity related extension

### PR DESCRIPTION
**Pull request by Awea**
Hi there,

This pull request:
- Fix a JSON error in extensions.json
- Add https://github.com/TobiahZ/unity-tools (under MIT license)
- Add https://github.com/Unity-Technologies/vscode-unity-debug (under MIT license)

### CODE REVIEW

File: extensions.json [hunk 1]
```diff
@@ -1110,6 +1110,11 @@
       "version": "0.10.1",
       "checkout": "v0.10.1"
     },
+    {
+      "id": "Tobiah.unity-tools",
+      "repository": "https://github.com/TobiahZ/unity-tools",
+      "version": "1.2.12"
```
Comment (spoenemann) [inline extensions.json:1116]:
```suggestion
      "version": "1.2.12",
      "checkout": "v1.2.12"
```

File: extensions.json [hunk 2]
```diff
@@ -1130,6 +1135,11 @@
       "id": "twxs.cmake",
       "repository": "https://github.com/twxs/vs.language.cmake"
     },
+    {
+      "id": "Unity.unity-debug",
+      "repository": "https://github.com/Unity-Technologies/vscode-unity-debug",
+      "version": "3.0.0"
```
Comment (spoenemann) [inline extensions.json:1141]:
```suggestion
      "version": "3.0.0",
      "checkout": "Version-3.0.1"
```
The mismatch is a bit strange here, though.

**Review by spoenemann — COMMENTED** [review]
Thanks for the contribution! I'd propose to add `checkout` properties to ensure that we build exactly the released versions.

**Review by jankeromnes — APPROVED** [review]
Many thanks for your contributions @Awea!

I took the liberty of accepting @spoenemann's suggestions, and this PR now looks good to go. 🚀

**Comment by Awea** [comment]
@jankeromnes thank you for handling my contributions and thank you @spoenemann for improving it :)

**Comment by Awea** [comment]
Hi there! Extensions aren't available on open-vsx.org, the publication process did not work?

**Comment by jankeromnes** [comment]
@Awea Oops, good catch.

In the [latest publish run logs](https://github.com/open-vsx/publish-extensions/runs/1388478556?check_suite_focus=true), we can indeed see that re-packaging the extension `Unity.unity-debug` failed with:

```
Running: git clone --recurse-submodules https://github.com/Unity-Technologies/vscode-unity-debug /tmp/repository
Running: git checkout Version-3.0.1
Running: npm install
Executing prepublish script 'npm run vscode:prepublish'...

> unity-debug@3.0.0 vscode:prepublish /tmp/repository
> make build

make ℹ info Invoking build target
Error: typescript/exceptions.ts(122,9): error TS2611: 'label' is defined as a property in class 'TreeItem', but is overridden here in 'Exception' as an accessor.
Error: typescript/exceptions.ts(130,9): error TS2611: 'tooltip' is defined as a property in class 'TreeItem', but is overridden here in 'Exception' as an accessor.
sh: 2: @echo: not found
make ✖ ERR  tsc -p ./typescript
@echo "build finished"
make ✖ ERR  Recipe exited with code %d
(node:21093) UnhandledPromiseRejectionWarning: Error: Recipe exited with code %d
```

Maybe we shouldn't attempt to re-package this extension ourselves. Since Unity provides a public `.vsix` file, we can simply re-publish that to OpenVSX.

Could you please remove the current `extensions.json` entry for `Unity.unity-debug`, and publish the `.vsix` file instead? Hint: You can add it automatically by running this command:

```
node add-extension --download=https://github.com/Unity-Technologies/vscode-unity-debug/releases/download/Version-2.7.2/unity-debug-2.7.2.vsix
```

Also, the other extension just failed because `Unity.unity-debug` wasn't published yet:

```
[FAIL] Could not process extension: {
  "id": "Tobiah.unity-tools",
  "repository": "https://github.com/TobiahZ/unity-tools",
  "version": "1.2.12",
  "checkout": "v1.2.12"
}
Error: Cannot resolve bundled extension: Unity.unity-debug
```

This should sort itself out automatically once `Unity.unity-debug` is successfully published.

**Pull request merged**